### ISSUE TITLE
The update of actor in DDPG

### ISSUE THREAD

**Issue by Exasor** [thu-ml/tianshou#277]
- [ ] I have marked all applicable categories:
    + [ ] exception-raising bug
    + [ ] RL algorithm bug
    + [ ] documentation request (i.e. "X is missing from the documentation.")
    + [x] new feature request
- [x] I have visited the [source website](https://github.com/thu-ml/tianshou/)
- [x] I have searched through the [issue tracker](https://github.com/thu-ml/tianshou/issues) for duplicates
- [ ] I have mentioned version numbers, operating system and environment, where applicable:
  ```python
  import tianshou, torch, sys
  print(tianshou.__version__, torch.__version__, sys.version, sys.platform)
  ```
Hi, I'm new in RL and find that the update of actor in DDPG as follows:
```
        actor_loss = -self.critic(batch.obs, action).mean()
        self.actor_optim.zero_grad()
        actor_loss.backward()
        self.actor_optim.step()
```
which is based on  the first line of the following equation:
![image](https://user-images.githubusercontent.com/56293255/105051736-07660a00-5aaa-11eb-9222-8caefa148de4.png)
but I want to make some modification in the gradient of policy which is the second item in the second line, and I searched many repositories but they are all use the same method as tianshou, so I want to know how to update actor using the equation of the second line in the former picture? Sincerely hope for your help.

**Comment by Trinkle23897** [comment]
The two equations are exactly the same, that's the chain rule. Actually we currently use the second equation: the first part is the critic net's gradient, and the second part is the gradient of actor net.

**Comment by Exasor** [comment]
Thanks for your reply,  I know that it's chain rule that connects these 2 equations, but I wonder if there are some ways to calculate the 2 gradient separately and combine them to update actor in pytorch, just like the following tensorflow code, because I want to make some modification in the second part, the gradient of actor net 
```
with tf.variable_scope('a_grad'):
    self.a_grads = tf.gradients(self.q, self.a)[0]
with tf.variable_scope('policy_grads'):
    self.policy_grads = tf.gradients(
        ys=self.a, xs=self.e_params, 
        grad_ys=a_grads 
    )
with tf.variable_scope('A_train'):
    opt = tf.train.AdamOptimizer(-self.lr) 
    self.train_op = opt.apply_gradients(zip(self.policy_grads, self.e_params))
```

**Comment by Trinkle23897** [comment]
How about using two separated optimizers for this equation? One for only actor part and another one for only critic part.
